lib/rpmsg/rpmsg_virtio.c

```diff
@@ -313,37 +313,55 @@ static int _rpmsg_virtio_get_buffer_size(struct rpmsg_virtio_device *rvdev)
 	return length;
 }
 
-static void rpmsg_virtio_hold_rx_buffer(struct rpmsg_device *rdev, void *rxbuf)
+static void rpmsg_virtio_hold_rx_buffer_nolock(struct rpmsg_device *rdev,
+					       struct rpmsg_hdr *rp_hdr)
 {
-	struct rpmsg_hdr *rp_hdr;
-
 	(void)rdev;
 
-	rp_hdr = RPMSG_LOCATE_HDR(rxbuf);
-
 	/* Set held status to keep buffer */
-	rp_hdr->reserved |= RPMSG_BUF_HELD;
+	rp_hdr->reserved += 1 << RPMSG_BUF_HELD_SHIFT;
```

Comment:
What about using macro ?
```c
/* Increase held status to keep buffer */
VIRTIO_BUFF_LOCK_INC( rp_hdr) rp_hdr->reserved += 1 << RPMSG_BUF_HELD_SHIFT

/* Dncrease held status to keep buffer */
VIRTIO_BUFF_LOCK_DEC( rp_hdr) rp_hdr->reserved -= 1 << RPMSG_BUF_HELD_SHIFT
```

Reply:
Done, and I change `VIRTIO_BUFF` to `RPMSG_BUF`